src/app/api/api.effects.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect } from '@ngrx/effects';
+
+import { filter, map, tap } from 'rxjs/operators';
+import { SnackBarService } from '../modules/shared/snack-bar.service';
+
+
+@Injectable()
+export class ApiEffects {
+
+  showApiErrorMessage$ = createEffect(() => this.actions$.pipe(
+    map((action: unknown) => (action as { errorMessageKey?: string }).errorMessageKey),
+    filter((errorMessageKey) => !!errorMessageKey),
+    tap((errorMessageKey) => this.snackBar.open(errorMessageKey as string))
+  ), { dispatch: false });
+
```

Comment:
Co tu sie do chuja odjaniepawla?

Reply:
:smile: 
operator 'as' to rzutowanie. 
` tap((errorMessageKey) => this.snackBar.open(errorMessageKey as string))`
Tutaj errorMessageKey jest typu "string | undefined". Linijkę wyżej odfiltrowujemy undefined, więc mogłem sobie to spokojnie zrzutować na sam string.

Wyrzej użyłem trochę większego hacka xd action jest typu Action, a taki typ nie ma pola errorMessageKey. Żeby sprawdzić, czy takie pole w akcji istnieje, wykorzystałem fakt, że unknown można zrzutować na wszystko ;d Więc najpierw przyjmowałem, że action jest typu unknown, a potem rzutowałem to na {erroMessageKey?: string}. Trochę pozmieniałem te kody, żeby były bardziej czytelne.

`
    map((action: Partial<ApiAction>) => action.errorMessageKey),
    filter((errorMessageKey) => !!errorMessageKey),
    tap((errorMessageKey: string | undefined) => this.snackBar.open(errorMessageKey as string))
`